pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -305,7 +305,7 @@ private void resumeReceivingFromPausedConsumersIfNeeded() {
                     break;
                 }
 
-                client.getInternalExecutorService().execute(() -> {
+                internalPinnedExecutor.execute(() -> {
```

Comment:
This seems to have changed the previous behavior. Before, you can select a different thread to execute each time, and now it has become a thread of execution.

Reply:
Yes I mentioned this in my first comment on the suggested approach. If we're going to avoid all multi-threaded calls to to `pendingBatchReceives.poll()' then we can't use the internal pool here, we must use only one.